internal/preflight/preflight.go

```diff
@@ -77,12 +78,12 @@ func (p *preflight) PreflightUpgradeCheck(ctx context.Context, pv *kmmv1beta1.Pr
 		log.Info(utils.WarnString("failed to update the stage of Module CR in preflight to build stage"), "module", mod.Name, "error", err)
 	}
 
-	return p.helper.verifyBuild(ctx, mapping, mod, kernelVersion)
+	return p.helper.verifyBuild(ctx, pv, mapping, mod)
 }
 
 type preflightHelperAPI interface {
 	verifyImage(ctx context.Context, mapping *kmmv1beta1.KernelMapping, mod *kmmv1beta1.Module, kernelVersion string) (bool, string)
-	verifyBuild(ctx context.Context, mapping *kmmv1beta1.KernelMapping, mod *kmmv1beta1.Module, kernelVersion string) (bool, string)
+	verifyBuild(ctx context.Context, pv *kmmv1beta1.PreflightValidation, mapping *kmmv1beta1.KernelMapping, mod *kmmv1beta1.Module) (bool, string)
```

Comment:
Should we replace the kernelVersion with pv in `verifyImage` as well?

Reply:
for now - no, i don't need pv there. As i wrote above, once Chris PR is in and i know for sure how signing is implemented, i will probably refactor the APIs